build/media_source/plg_system_shortcut/keyselect.es6.js

```diff
@@ -0,0 +1,81 @@
+class JoomlaShortcutModal {
+	constructor() {
+		
+		if (!Joomla) {
+			throw new Error('Joomla API is not properly initialised');
+		}
+		// Bindings
+    	this.setAttributes = this.setAttributes.bind(this);
+		this.buildKeySelectModal = this.buildKeySelectModal.bind(this);
+
+		document.addEventListener("DOMContentLoaded",this.buildKeySelectModal);
+		
+		var keySelectBtns = document.getElementsByClassName("keySelectBtn");
+		for(var x = 0; x < keySelectBtns.length; x++) {
+			keySelectBtns[x].addEventListener("click",this.handleClickEvent,false);
+		}
+		
+	}
+	
+	buildKeySelectModal() {
+		
+		
+		const modal_div = document.createElement("div");
+		this.setAttributes(modal_div, {"class": "modal fade", "id": "keySelectModal", "tabindex": "-1", "role": "dialog", "aria-labelledby": "keySelectModalLabel", "aria-hidden": "true"});
+		modal_div.innerHTML = '<div class="modal-dialog" role="document"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="keySelectModalLabel">'+Joomla.getOptions('modal_header_title')+'</h5><button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button></div><div class="modal-body"><div class="p-3"><p>'+Joomla.getOptions('modal_combination_text')+' <span id="currentKeyCombination"></span></p><div class="form-group"><input id="keyCombination" class="form-control" type="text" autocomplete="off" maxlength="1" onkeypress="return /[a-z]/i.test(event.key)" /><input type="hidden" id="current_KeyEvent" value="" /></div></div></div><div class="modal-footer"  <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button> <button type="button" class="btn btn-primary">Save changes</button> </div></div></div>';
+		document.body.appendChild(modal_div); 
+		Joomla.initialiseModal(keySelectModal);
+		const keyCombination = document.getElementById("keyCombination");
```

Comment:
i guess no need to use the input field 

check this  https://developer.mozilla.org/en-US/docs/Web/API/Document/keydown_event